posts/2021-11-09-now-with-more-elixir.md

```diff
@@ -0,0 +1,208 @@
+      <pubDate>#{Calendar.strftime(post.date, "%a, %d %B %Y 00:00:00 +0000")}</pubDate>
+      <link>#{link}</link>
+      <guid isPermaLink="true">#{link}</guid>
+    </item>
+    """
+  end
+end
+```
+
+The sitemap generator follows a similar format but the value of content and its structure makes the task a bit more complicated. We break down the generation into a couple of steps:
+
+1. Add the blog index and our privacy policy
+
+   ```elixir
+   defp all_links do
+     [
+       Helpers.post_url(Endpoint, :index),
+       Helpers.page_url(Endpoint, :privacy)
+     ] ++ post_links() ++ Enum.flat_map(supported_locales(), &locale_links/1)
+   end
+   ```
+
+2. Create a collection of blog post links
+
+   ```elixir
+   defp post_links do
+     0..(Posts.pages() - 1)
+     |> Enum.flat_map(&Posts.page/1)
+     |> Enum.map(&Helpers.post_url(Endpoint, :show, &1.slug))
+   end
+   ```
+
+3. Build up the links for each locale. This includes all lessons and pages such as conferences, podcasts, "Why Elixir?", and others.
+
+   ```elixir
+   defp locale_links(locale), do: page_links(locale) ++ lesson_links(locale)
+
+   defp page_links(locale) do
+     [
+       Helpers.page_url(Endpoint, :conferences, locale),
+       Helpers.page_url(Endpoint, :index, locale),
+       Helpers.page_url(Endpoint, :podcasts, locale),
+       Helpers.page_url(Endpoint, :why, locale),
+       Helpers.report_url(Endpoint, :index, locale)
+     ]
+   end
+
+   defp lesson_links(locale) do
+     config = Application.get_env(:school_house, :lessons)
+
+     translated_lesson_links =
+       for {section, lessons} <- config, lesson <- lessons, translated_lesson?(section, lesson, locale) do
+       Helpers.lesson_url(Endpoint, :lesson, section, lesson, locale)
+       end
+
+     section_indexes =
+       for section <- Keyword.keys(config) do
+       Helpers.lesson_url(Endpoint, :index, section, locale)
+       end
+
+     section_indexes ++ translated_lesson_links
+   end
+
+   defp translated_lesson?(section, lesson, locale) do
+     case Lessons.get(section, lesson, locale) do
+     {:ok, _} -> true
+     _ -> false
+     end
+   end
+
+   defp supported_locales do
+     :school_house
+     |> Application.get_env(SchoolHouseWeb.Gettext)
+     |> Keyword.get(:locales)
+   end
+   ```
+
+Both tasks are run in our release process and stored in the static directory to be served alongside things like `robots.txt`:
+
+```elixir
+plug Plug.Static,
+  at: "/",
+  from: :school_house,
+  gzip: false,
+  only: ~w(css fonts images js favicon.ico robots.txt feed.xml sitemap.xml)
+```
+
+### Roadmap Forward
+
+#### New Content
+
+With this launch we also restructured our content into new sections! This paves the way for much more content and a more managable way to organize and consume it. Some of the new content you can be on the look out for:
+
+1. **Advanced Content**
+
+   1. Dedicated meta programming lesson that explores this powerful functionality
```

Comment:
```suggestion
   1. Dedicated metaprogramming lesson that explores this powerful functionality
```

Wikipedia (and others) seem to think this should be one word (https://www.google.com/search?q=metaprogramming) , but my browser thinks it is spelled incorrectly, so I'm not sure which way to go on this